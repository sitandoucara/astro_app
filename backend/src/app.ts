import express from "express";
import dotenv from "dotenv";
import cors from "cors";
import chartRoutes from "./routes/chartRoutes";
import timezoneRoutes from "./routes/timezoneRoutes";

dotenv.config();

const app = express();
app.use(cors());
app.use(express.json());

app.use("/api/chart", chartRoutes);
app.use("/api/timezone", timezoneRoutes);

const PORT = process.env.PORT || 4000;
app.listen(PORT, () => {
  console.log(`Server running on port ${PORT}`);
});
