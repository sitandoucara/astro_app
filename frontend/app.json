{
  "expo": {
    "name": "my-expo-app",
    "slug": "my-expo-app",
    "version": "1.0.9",
    "web": {
      "bundler": "metro",
      "favicon": "./assets/favicon.png"
    },
    "experiments": {
      "tsconfigPaths": true
    },
    "plugins": ["expo-font"],
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "userInterfaceStyle": "light",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "assetBundlePatterns": ["**/*"],
    "ios": {
      "supportsTablet": false,
      "bundleIdentifier": "com.sigraph.astromood",
      "infoPlist": {
        "ITSAppUsesNonExemptEncryption": false
      }
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/adaptive-icon.png",
        "backgroundColor": "#ffffff"
      }
    }
  }
}
