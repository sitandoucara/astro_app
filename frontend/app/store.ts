import AsyncStorage from '@react-native-async-storage/async-storage';
import { configureStore } from '@reduxjs/toolkit';
import { persistStore, persistReducer } from 'redux-persist';

import authReducer from '../features/auth/AuthSlice';
import languageReducer from '../shared/language/languageSlice';
import themeReducer from '../shared/theme/themeSlice';

const themePersistConfig = {
  key: 'theme',
  storage: AsyncStorage,
};

const languagePersistConfig = {
  key: 'language',
  storage: AsyncStorage,
};

const persistedThemeReducer = persistReducer(themePersistConfig, themeReducer);
const persistedLanguageReducer = persistReducer(languagePersistConfig, languageReducer);

export const store = configureStore({
  reducer: {
    auth: authReducer,
    theme: persistedThemeReducer,
    language: persistedLanguageReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST', 'persist/REHYDRATE'],
      },
    }),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
