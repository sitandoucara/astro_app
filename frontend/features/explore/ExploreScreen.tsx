import { FontAwesome, MaterialCommunityIcons } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { RootStackParamList } from 'navigation/types';
import { useLayoutEffect } from 'react';
import { Text, TouchableOpacity, View, Alert } from 'react-native';
import { useAppSelector } from 'shared/hooks';
import { useThemeColors } from 'shared/hooks/useThemeColors';

export default function ExploreScreen() {
  const isDarkMode = useAppSelector((state) => state.theme.isDarkMode);
  const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  const colors = useThemeColors();

  useLayoutEffect(() => {
    navigation.setOptions({
      headerTitle: () => (
        <Text className="text-aref ml-5 text-[20px]" style={{ color: colors.textColor }}>
          You can Explore
        </Text>
      ),
      headerTitleAlign: 'left',
    });
  }, [navigation, isDarkMode]);

  return (
    <View
      className="flex-1 items-center justify-center px-2"
      style={{ backgroundColor: colors.backgroundColor }}>
      {/* Learn */}
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => navigation.navigate('LearnScreen')}
        className={`w-full rounded-3xl p-6 ${colors.cardBg} border ${colors.borderColor} mb-4`}>
        <View className="justify-center">
          <View className="h-16 w-16 items-center justify-center">
            <View
              className={`mr-3 h-10 w-10 items-center justify-center rounded-full ${colors.iconBg}`}>
              <FontAwesome name="book" size={20} style={{ color: colors.iconColorAlt }} />
            </View>
          </View>

          <View>
            <Text className={`text-aref text-xl font-semibold ${colors.textPrimary} mb-2`}>
              Learn
            </Text>
            <Text className={`text-aref text-sm ${colors.textSecondaryAlt}`}>
              Discover Astrology
            </Text>
          </View>
        </View>
      </TouchableOpacity>

      {/* Test & Quiz */}
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => navigation.navigate('QuizzScreen')}
        className={`w-full rounded-3xl p-6 ${colors.cardBg} border ${colors.borderColor} mb-4`}>
        <View className="justify-center">
          <View className="h-16 w-16 items-center justify-center">
            <View
              className={`mr-3 h-10 w-10 items-center justify-center rounded-full ${colors.iconBg}`}>
              <MaterialCommunityIcons
                name="gamepad-variant"
                size={20}
                style={{ color: colors.iconColorAlt }}
              />
            </View>
          </View>
          <View>
            <Text className={`text-aref text-xl font-semibold ${colors.textPrimary} mb-2`}>
              Test & Quiz
            </Text>
            <Text className={`text-aref text-sm ${colors.textSecondaryAlt}`}>
              Test Your Astrology Knowlege
            </Text>
          </View>
        </View>
      </TouchableOpacity>

      {/*Mini books */}
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => Alert.alert('Zodiac Sign Compatibility clicked!')}
        className={`w-full rounded-3xl p-6 ${colors.cardBg} border ${colors.borderColor} mb-4`}>
        <View className="justify-center">
          <View className="h-16 w-16 items-center justify-center">
            <View
              className={`mr-3 h-10 w-10 items-center justify-center rounded-full ${colors.iconBg}`}>
              <MaterialCommunityIcons
                name="book"
                size={20}
                style={{ color: colors.iconColorAlt }}
              />
            </View>
          </View>

          <View>
            <Text className={`text-aref text-xl font-semibold ${colors.textPrimary} mb-2`}>
              Mini-books
            </Text>
            <Text className={`text-aref text-sm ${colors.textSecondaryAlt}`}>
              Know more about Astrology
            </Text>
          </View>
        </View>
      </TouchableOpacity>
    </View>
  );
}
