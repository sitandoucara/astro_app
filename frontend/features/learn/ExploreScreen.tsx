import { FontAwesome, MaterialCommunityIcons } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { RootStackParamList } from 'navigation/types';
import { useLayoutEffect } from 'react';
import { Text, TouchableOpacity, View, Alert } from 'react-native';
import { useAppSelector } from 'shared/hooks';

export default function ExploreScreen() {
  const isDarkMode = useAppSelector((state) => state.theme.isDarkMode);
  const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  const backgroundColor = isDarkMode ? '#F2EAE0' : '#281109';
  const textColor = isDarkMode ? '#32221E' : '#F2EAE0';
  const iconBg = isDarkMode ? 'bg-light-border' : 'bg-dark-border';
  const iconColor = isDarkMode ? '#F2EAE0' : '#32221E';
  const cardBg = isDarkMode ? 'bg-light-cardback' : 'bg-[#442F29]/50';
  const borderColor = isDarkMode ? 'border-light-border' : 'border-dark-border';
  const textPrimary = isDarkMode ? 'text-light-text1' : 'text-dark-text1';
  const textSecondary = isDarkMode ? 'text-[#D8D3D0]' : 'text-[#D9D5D4]';

  useLayoutEffect(() => {
    navigation.setOptions({
      headerTitle: () => (
        <Text className="text-aref ml-5 text-[20px]" style={{ color: textColor }}>
          You can Explore
        </Text>
      ),
      headerTitleAlign: 'left',
    });
  }, [navigation, isDarkMode]);

  return (
    <View className="flex-1 items-center justify-center px-2" style={{ backgroundColor }}>
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => navigation.navigate('LearnScreen')}
        className={` w-full rounded-3xl p-6 ${cardBg} border ${borderColor} mb-4`}>
        <View className="  justify-center">
          <View className="h-16 w-16 items-center justify-center ">
            <View className={`mr-3 h-10 w-10 items-center justify-center rounded-full  ${iconBg}`}>
              <FontAwesome name="book" size={20} style={{ color: iconColor }} />
            </View>
          </View>

          <View className="">
            <Text className={`text-aref text-xl font-semibold ${textPrimary} mb-2`}>Learn</Text>
            <Text className={`text-aref text-sm ${textSecondary}`}>Discover Astrology</Text>
          </View>
        </View>
      </TouchableOpacity>

      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => navigation.navigate('QuizzScreen')}
        className={` w-full rounded-3xl p-6 ${cardBg} border ${borderColor} mb-4`}>
        <View className="  justify-center">
          {/* Header avec les 2 grandes icones */}
          <View className="h-16 w-16 items-center justify-center ">
            <View className={`mr-3 h-10 w-10 items-center justify-center rounded-full  ${iconBg}`}>
              <MaterialCommunityIcons
                name="gamepad-variant"
                size={20}
                style={{ color: iconColor }}
              />
            </View>
          </View>

          {/* Texte centré */}
          <View className="">
            <Text className={`text-aref text-xl font-semibold ${textPrimary} mb-2`}>
              Test & Quiz
            </Text>
            <Text className={`text-aref text-sm ${textSecondary}`}>
              Test Your Astrology Knowlege
            </Text>
          </View>
        </View>
      </TouchableOpacity>

      <TouchableOpacity
        activeOpacity={0.8}
        onPress={() => Alert.alert('Zodiac Sign Compatibility clicked!')}
        className={` w-full rounded-3xl p-6 ${cardBg} border ${borderColor} mb-4`}>
        <View className="  justify-center">
          {/* Header avec les 2 grandes icones */}
          <View className="h-16 w-16 items-center justify-center ">
            <View className={`mr-3 h-10 w-10 items-center justify-center rounded-full  ${iconBg}`}>
              <MaterialCommunityIcons name="book" size={20} style={{ color: iconColor }} />
            </View>
          </View>

          {/* Texte centré */}
          <View className="">
            <Text className={`text-aref text-xl font-semibold ${textPrimary} mb-2`}>
              Mini-books
            </Text>
            <Text className={`text-aref text-sm ${textSecondary}`}>Know more about Astrology</Text>
          </View>
        </View>
      </TouchableOpacity>
    </View>
  );
}
