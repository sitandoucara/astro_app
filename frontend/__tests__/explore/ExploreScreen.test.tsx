import { NavigationContainer } from '@react-navigation/native';
import { render, fireEvent } from '@testing-library/react-native';
import { store } from 'app/store';
import ExploreScreen from 'features/explore/ExploreScreen';
import React from 'react';
import { Alert } from 'react-native';
import { Provider } from 'react-redux';

// MOCK ICONS TO AVOID `displayName` / NATIVE ERRORS
jest.mock('@expo/vector-icons', () => {
  const React = require('react');
  return {
    FontAwesome: (props: any) => <></>,
    MaterialCommunityIcons: (props: any) => <></>,
  };
});

// MOCK NAVIGATION
const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockNavigate,
      setOptions: jest.fn(),
    }),
  };
});

// RENDER FUNCTION WITH STORE + NAVIGATION
const renderWithProviders = (component: React.ReactElement) => {
  return render(
    <Provider store={store}>
      <NavigationContainer>{component}</NavigationContainer>
    </Provider>
  );
};

describe('ExploreScreen Tests', () => {
  it('A - Explore screen renders correctly (snapshot)', () => {
    const { toJSON } = renderWithProviders(<ExploreScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("B - 'Learn' button navigates to LearnScreen", () => {
    const { getByText } = renderWithProviders(<ExploreScreen />);
    fireEvent.press(getByText('Learn'));
    expect(mockNavigate).toHaveBeenCalledWith('LearnScreen');
  });

  it("C - 'Test & Quiz' button navigates to QuizzScreen", () => {
    const { getByText } = renderWithProviders(<ExploreScreen />);
    fireEvent.press(getByText('Test & Quiz'));
    expect(mockNavigate).toHaveBeenCalledWith('QuizzScreen');
  });

  it("D - 'Mini-books' button triggers an alert", () => {
    const alertMock = jest.spyOn(Alert, 'alert').mockImplementation(() => {});
    const { getByText } = renderWithProviders(<ExploreScreen />);
    fireEvent.press(getByText('Mini-books'));
    expect(alertMock).toHaveBeenCalledWith('Zodiac Sign Compatibility clicked!');
    alertMock.mockRestore();
  });
});
