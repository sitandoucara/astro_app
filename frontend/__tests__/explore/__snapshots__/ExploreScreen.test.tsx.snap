// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreScreen Tests A - Explore screen renders correctly (snapshot) 1`] = `
<View
  className="flex-1 items-center justify-center px-2"
  style={
    {
      "backgroundColor": "#281109",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      className="  justify-center"
    >
      <View
        className="h-16 w-16 items-center justify-center "
      >
        <View
          className="mr-3 h-10 w-10 items-center justify-center rounded-full  bg-dark-border"
        />
      </View>
      <View>
        <Text
          className="text-aref text-xl font-semibold text-dark-text1 mb-2"
        >
          Learn
        </Text>
        <Text
          className="text-aref text-sm text-[#D9D5D4]"
        >
          Discover Astrology
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      className="justify-center"
    >
      <View
        className="h-16 w-16 items-center justify-center "
      >
        <View
          className="mr-3 h-10 w-10 items-center justify-center rounded-full  bg-dark-border"
        />
      </View>
      <View>
        <Text
          className="text-aref text-xl font-semibold text-dark-text1 mb-2"
        >
          Test & Quiz
        </Text>
        <Text
          className="text-aref text-sm text-[#D9D5D4]"
        >
          Test Your Astrology Knowlege
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      className="justify-center"
    >
      <View
        className="h-16 w-16 items-center justify-center "
      >
        <View
          className="mr-3 h-10 w-10 items-center justify-center rounded-full  bg-dark-border"
        />
      </View>
      <View>
        <Text
          className="text-aref text-xl font-semibold text-dark-text1 mb-2"
        >
          Mini-books
        </Text>
        <Text
          className="text-aref text-sm text-[#D9D5D4]"
        >
          Know more about Astrology
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ExploreScreen Tests A - L'Ã©cran Explore s'affiche sans crash (snapshot) 1`] = `
<View
  className="flex-1 items-center justify-center px-2"
  style={
    {
      "backgroundColor": "#281109",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      className="  justify-center"
    >
      <View
        className="h-16 w-16 items-center justify-center "
      >
        <View
          className="mr-3 h-10 w-10 items-center justify-center rounded-full  bg-dark-border"
        />
      </View>
      <View>
        <Text
          className="text-aref text-xl font-semibold text-dark-text1 mb-2"
        >
          Learn
        </Text>
        <Text
          className="text-aref text-sm text-[#D9D5D4]"
        >
          Discover Astrology
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      className="justify-center"
    >
      <View
        className="h-16 w-16 items-center justify-center "
      >
        <View
          className="mr-3 h-10 w-10 items-center justify-center rounded-full  bg-dark-border"
        />
      </View>
      <View>
        <Text
          className="text-aref text-xl font-semibold text-dark-text1 mb-2"
        >
          Test & Quiz
        </Text>
        <Text
          className="text-aref text-sm text-[#D9D5D4]"
        >
          Test Your Astrology Knowlege
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      className="justify-center"
    >
      <View
        className="h-16 w-16 items-center justify-center "
      >
        <View
          className="mr-3 h-10 w-10 items-center justify-center rounded-full  bg-dark-border"
        />
      </View>
      <View>
        <Text
          className="text-aref text-xl font-semibold text-dark-text1 mb-2"
        >
          Mini-books
        </Text>
        <Text
          className="text-aref text-sm text-[#D9D5D4]"
        >
          Know more about Astrology
        </Text>
      </View>
    </View>
  </View>
</View>
`;
